apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-2572-Cleanup-Medicare-Beneficairyid-History/invoke-psql.sh

```diff
@@ -0,0 +1,346 @@
+  BENE_BIRTH_DT,
+  LAST_UPDATED)
+SELECT
+  NEXTVAL('beneficiaryhistory_beneficiaryhistoryid_seq'),
+  V2.BENE_ID,
+  V2.MBI_NUM,
+  V2.EFCTV_BGN_DT,
+  V2.EFCTV_END_DT,
+  X.BENE_CRNT_HIC_NUM,
+  X.HICN_UNHASHED,
+  X.BENE_SEX_IDENT_CD,
+  X.BENE_BIRTH_DT,
+  NOW()
+FROM
+  (SELECT TBL1.BENE_MBI_ID,
+      TBL1.BENE_ID,
+      TBL1.MBI_NUM,
+      TBL1.MBI_EFCTV_BGN_DT AS EFCTV_BGN_DT,
+      TBL1.MBI_EFCTV_END_DT AS EFCTV_END_DT,
+      TBL1.LAST_UPDATED
+    FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP TBL1
+    WHERE NOT EXISTS
+        (SELECT V1.*
+          FROM
+            (SELECT DISTINCT
+                A.BENE_ID,
+                A.MBI_NUM,
+                A.EFCTV_BGN_DT
+              FROM BENEFICIARIES_HISTORY A
+              WHERE A.BENE_ID in
+                  (SELECT DISTINCT BENE_ID
+                    FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP)
+                AND A.EFCTV_BGN_DT IS NOT NULL ) V1
+          WHERE TBL1.BENE_ID = V1.BENE_ID
+            AND TBL1.MBI_NUM = V1.MBI_NUM
+            AND V1.EFCTV_BGN_DT IS NOT NULL
+            AND TBL1.MBI_EFCTV_BGN_DT = V1.EFCTV_BGN_DT ) ) V2
+INNER JOIN BENEFICIARIES X ON X.BENE_ID = V2.BENE_ID
+WHERE X.MBI_NUM <> V2.MBI_NUM
+ORDER BY 2, 3;
+EOM
+
+# ==============================
+# BEGIN PROCESSING
+# ==============================
+# belt-and-suspenders to check connection to database.
+echo "Testing db connectivity..."
+now=$(psql -h "${PGHOST}" -U "${PGUSER}" -d "${PGDATABASE}" --quiet --tuples-only -c "select NOW();")
+if [[ "$now" == *"202"* ]]; then
+  echo "db connectivity: OK"
+else
+  echo "db connectivity: FAILED...exiting"
+  exit 1;
+fi
+
+# STEP 1
+# --------------------------------------------------------------------
+# Get a count of records in the MEDICARE_BENEFICIARYID_HISTORY table.
+# --------------------------------------------------------------------
+echo "checking current record counts..."
+SQL=$(printf "%s\n\n%s" "SELECT COUNT(*) FROM MEDICARE_BENEFICIARYID_HISTORY")
+CNT=$(psql -h "${PGHOST}" -U "${PGUSER}" -d "${PGDATABASE}" --tuples-only -c "${SQL}")
+if [ -n "${CNT}" ]; then
+  now=$(date +'%T')
+  printf "%s : Start count of records in MEDICARE_BENEFICIARYID_HISTORY : %d\n\n" "${now}" "${CNT}";
+fi
+SQL=$(printf "%s\n\n%s" "SELECT COUNT(*) FROM BENEFICIARIES_HISTORY")
+CNT=$(psql -h "${PGHOST}" -U "${PGUSER}" -d "${PGDATABASE}" --tuples-only -c "${SQL}")
+if [ -n "${CNT}" ]; then
+  now=$(date +'%T')
+  printf "%s : Start count of records in BENEFICIARIES_HISTORY : %d\n\n" "${now}" "${CNT}";
+fi
+
+# STEP 2
+# -------------------------------------------------------------------
+# Create a temp MEDICARE_BENEFICIARYID_HISTORY_TEMP table that we'll
+# use for processing; we do it this way as we'll be doing some destructive
+# operations (DELETEs) as we go through the various steps.
+# -------------------------------------------------------------------
+echo "Begin processing of Step 2 at: $(date +'%T')"
+echo "${STEP_TWO_SETUP_TEMP_SQL}"
```

Comment:
Should this be wrapped in if `[ "${SHOW_SQL}" = true ] ; then`?

Reply:
Probably not worth it; I only had the SHOW_SQL for stuff that really did something; in those places you can cut-paste the SQL, remove things like the DELETE FROM and replace with SELECT FROM and you can QC both the sql and the result set. For create / select it's just some menial sql for that and not worth anything.

Reply:
should it not be echo'd out then?